main.js

```diff
@@ -0,0 +1,58 @@
+class Color {
+  constructor() {
+    this.hexcode = this.#randomHexCode();
+    this.locked = false;
+    this.id = Color.colorsMadeSoFar + 1;
+    Color.colorsMadeSoFar++
+  }
+
+  static colorsMadeSoFar = 0;
+
+  #randomHexCode() {
+    var validCharacters = "ABCDEF0123456789".split("");
+
+    var output = ["#"];
+
+    for(var i = 0; i < 6; i++) {
+      var randomIndex = Math.floor(Math.random() * validCharacters.length);
+      output.push(validCharacters[randomIndex]);
+    }
+
+    return output.join('');
+  }
+}
+
+class Palette {
+  constructor() {
+    this.colors = this.#addDefaultColors();
+    this.id = Palette.palettesMadeSoFar + 1;
+    Palette.palettesMadeSoFar++;
+  }
+
+  static palettesMadeSoFar = 0;
+
+  #addDefaultColors() {
+    var output = [];
+    for(var i = 0; i < 5; i++) {
+      output.push(new Color());
+    }
+    return output;
+  }
+
+  lockColor(colorId) {
+    for(var i = 0; i < this.colors.length; i++) {
+      if(this.colors[i].id === colorId) {
+        this.colors[i].locked = true;
```

Comment:
I'd love to see us refactor this so that code in the Palette class doesn't reach into the Color and alter its properties directly.

i.e. in lockColor, we could write `this.colors[i].lock()` 
and in the Color class, we could add an instance method like so: `lock(){ this.locked = true }`

Reply:
This approach has a few benefits. It's related to a design pattern that has to do with the Single Responsibility Principal, making our programs more modular and re-usable, and making our code less brittle (i.e. harder to mess up with future code).

https://en.wikipedia.org/wiki/Encapsulation_(computer_programming)